xblock/test/test_mixins.py

```diff
@@ -260,38 +271,151 @@ def _assert_node_attributes(self, node, expected_attributes):
             else:
                 self.assertIsNotNone(node.get(key))
 
-    def test_add_xml_to_node(self):
-        """ Tests add_xml_to_node with various field defaults and runtime parameters """
-        block_type = self.TestXBlock
-        block = self._make_block(block_type)
-        node = etree.Element(block_type.etree_node_tag)
+    def _assert_node_elements(self, node, expected_elements, check_none_attr=True):
+        """
+        Checks XML node elements to match expected elements.
+        """
+        node_elements = list(node)
+        self.assertEqual(set([elem.tag for elem in node_elements]), set(expected_elements.keys()))
+        if check_none_attr:
+            # All elements on the node are expected to have a "none"="true" attribute.
+            for elem in node:
+                self.assertEqual(elem.get('none'), 'true')
```

Comment:
It seems like maybe there should be a separate `assert_node_nones` (or something) that does the assertion that's currently happening if `check_none_attr` is true. It seems odd to have an assertion function that takes an argument to do an optional additional assertion.